pkg/interfaces/protocoldriver.go

```diff
@@ -54,4 +54,14 @@ type ProtocolDriver interface {
 	// RemoveDevice is a callback function that is invoked
 	// when a Device associated with this Device Service is removed
 	RemoveDevice(deviceName string, protocols map[string]models.ProtocolProperties) error
+
+	// Discover triggers protocol specific device discovery, asynchronously
+	// writes the results to the channel which is passed to the implementation
+	// via ProtocolDriver.Initialize(). The results may be added to the device service
+	// based on a set of acceptance criteria (i.e. Provision Watchers).
+	Discover()
```

Comment:
add an error return to this function to determine whether the Device Service supports Discovery()

Reply:
Added error return.